docs_user/modules/proc_adopting-compute-services-to-the-data-plane.adoc

```diff
@@ -240,18 +331,60 @@ $ oc create secret generic redhat-registry \
 * Replace `<registry_username>` with the applicable user name.
 * Replace `<registry_password>` with the applicable password.
 endif::[]
++
+
+[NOTE]
+The `subscription-manager` secret does not need to be referenced in `OpenStackDataPlaneService`'s `spec.dataSources` data.
+It is already passed in via a node-specific `OpenStackDataPlaneNodeSet` data in `spec.nodeTemplate.ansible.ansibleVarsFrom`.
+
+
+. Create the dataplane node sets definitions for each cell:
 
-. Deploy the `OpenStackDataPlaneNodeSet` CR:
 +
 [source,yaml]
 ----
-$ oc apply -f - <<EOF
+$ declare -A names
+$ for CELL in $(echo $RENAMED_CELLS); do
+  ref="COMPUTES_$(echo ${CELL}|tr '[:lower:]' '[:upper:]')"
+  eval names=\${!${ref}[@]}
+  [ -z "$names" ] && continue
+  ind=0
+  rm -f computes-$CELL
+  for compute in $names; do
+    ip="${ref}['$compute']"
+    cat >> computes-$CELL << EOF
+    ${compute}:
+      hostName: $compute
+      ansible:
+        ansibleHost: $compute
+      networks: <1>
+      - defaultRoute: true
+        fixedIP: ${!ip}
+        name: ctlplane
+        subnetName: subnet1
+      - name: internalapi
+        subnetName: subnet1
+      - name: storage
+        subnetName: subnet1
+      - name: tenant
+        subnetName: subnet1
+EOF
+    ind=$(( ind + 1 ))
+  done
+
+  test -f computes-$CELL || continue
+  if [ "$CELL" = "cell1" ]; then
+    GLOBAL="- ssh-known-hosts"
+  else
+    GLOBAL=" "
+  fi
```

Comment:
hm, I think this is changed since https://github.com/openstack-k8s-operators/openstack-operator/commit/19f6d2cb49abbcd17f41f8fb5a2c2b8e81368bc6 So you can drop the conditional and include the ssh-known-hosts service to each NodeSet.